### PR TITLE
[AI] Pin bundle size baselines to the base commit

### PR DESCRIPTION
**Pull request by tim-smart**
## Summary

- resolve the bundle-size base artifact by the pull request's exact base SHA
- verify that every returned workflow run matches the requested commit before using its artifacts
- cover the stale-run regression and workflow wiring in the CI-actions tests

## Root cause

The base lookup filtered successful `build.yml` runs by branch and accepted the first result without checking its commit. In the size comparison for actualbudget/actual#8752, that request returned run `30768952310` from August 2 even though the pull request base SHA had successful run `32157684269` from August 18. The stale artifacts made unrelated changes on `master` appear to belong to the pull request.

## Validation

- `yarn workspace @actual-app/ci-actions test` (19 tests)
- `yarn workspace @actual-app/ci-actions typecheck`
- `yarn lint`
- confirmed the exact-SHA Actions API query resolves run `32157684269`

## AI usage

Implemented and tested with OpenAI Codex. The workflow behavior, GitHub Actions run evidence, diff, and test results were reviewed by the contributor.

<hr />

### Bundle Stats

Bundle | Files count | Total bundle size | % Changed
------ | ----------- | ----------------- | ---------
desktop-client | 42 | 13.53 MB | 0%
loot-core | 1 | 4.63 MB | 0%
api | 2 | 3.84 MB | 0%
cli | 1 | 8.02 MB | 0%
crdt | 1 | 11.16 kB | 0%

<details>
<summary>View detailed bundle stats</summary>

#### desktop-client

**Total**
Files count | Total bundle size | % Changed
----------- | ----------------- | ---------
42 | 13.53 MB | 0%

<details>
<summary>View detailed bundle breakdown</summary>
<div>

**Added**
No assets were added

**Removed**
No assets were removed

**Bigger**
No assets were bigger

**Smaller**
No assets were smaller

**Unchanged**
Asset | File Size | % Changed
----- | --------- | ---------
static/js/index.js | 1.57 MB | 0%
static/js/AppliedFilters.js | 35.51 kB | 0%
static/js/BackgroundImage.js | 121.09 kB | 0%
static/js/FormulaEditor.js | 766.28 kB | 0%
static/js/LoadingIndicator.js | 669.76 kB | 0%
static/js/ManageRules.js | 71.66 kB | 0%
static/js/PayeeRuleCountLabel.js | 12.81 kB | 0%
static/js/ReportRouter.js | 1.42 MB | 0%
static/js/ScheduleEditForm.js | 76.54 kB | 0%
static/js/SchedulesTable.js | 171.25 kB | 0%
static/js/TourHost.js | 169.22 kB | 0%
static/js/TourProvider.js | 1.54 kB | 0%
static/js/TransactionEdit.js | 219.94 kB | 0%
static/js/TransactionList.js | 79.22 kB | 0%
static/js/Value.js | 1.93 MB | 0%
static/js/bootstrapHyperFormula.js | 302 B | 0%
static/js/ca.js | 171.68 kB | 0%
static/js/client.js | 452.05 kB | 0%
static/js/de.js | 183.43 kB | 0%
static/js/en-GB.js | 11.1 kB | 0%
static/js/en.js | 245.47 kB | 0%
static/js/es.js | 165.11 kB | 0%
static/js/fr.js | 180.13 kB | 0%
static/js/indexeddb-main-thread-worker-e59fee74.js | 13.46 kB | 0%
static/js/it.js | 152.73 kB | 0%
static/js/months.js | 574.49 kB | 0%
static/js/narrow.js | 287.24 kB | 0%
static/js/nb-NO.js | 136.51 kB | 0%
static/js/nl.js | 152.45 kB | 0%
static/js/pl.js | 137.04 kB | 0%
static/js/pt-BR.js | 179.25 kB | 0%
static/js/ru.js | 209.49 kB | 0%
static/js/theme.js | 31.1 kB | 0%
static/js/uk.js | 200.72 kB | 0%
static/js/useDateFormat.js | 2.37 MB | 0%
static/js/useFormatList.js | 10.22 kB | 0%
static/js/useLocalPref.js | 97.2 kB | 0%
static/js/useReducedMotion.js | 594 B | 0%
static/js/useTransactionBatchActions.js | 11.03 kB | 0%
static/js/wide.js | 274.04 kB | 0%
static/js/workbox-window.prod.es5.js | 7.21 kB | 0%
static/js/zh-Hans.js | 107.54 kB | 0%
</div>
</details>

---

#### loot-core

**Total**
Files count | Total bundle size | % Changed
----------- | ----------------- | ---------
1 | 4.63 MB | 0%

<details>
<summary>View detailed bundle breakdown</summary>
<div>

**Added**
No assets were added

**Removed**
No assets were removed

**Bigger**
No assets were bigger

**Smaller**
No assets were smaller

**Unchanged**
Asset | File Size | % Changed
----- | --------- | ---------
kcab.worker.DOLEluIG.js | 4.63 MB | 0%
</div>
</details>

---

#### api

**Total**
Files count | Total bundle size | % Changed
----------- | ----------------- | ---------
2 | 3.84 MB | 0%

<details>
<summary>View detailed bundle breakdown</summary>
<div>

**Added**
No assets were added

**Removed**
No assets were removed

**Bigger**
No assets were bigger

**Smaller**
No assets were smaller

**Unchanged**
Asset | File Size | % Changed
----- | --------- | ---------
index.js | 3.84 MB | 0%
models.js | 0 B | 0%
</div>
</details>

---

#### cli

**Total**
Files count | Total bundle size | % Changed
----------- | ----------------- | ---------
1 | 8.02 MB | 0%

<details>
<summary>View detailed bundle breakdown</summary>
<div>

**Added**
No assets were added

**Removed**
No assets were removed

**Bigger**
No assets were bigger

**Smaller**
No assets were smaller

**Unchanged**
Asset | File Size | % Changed
----- | --------- | ---------
cli.js | 8.02 MB | 0%
</div>
</details>

---

#### crdt

**Total**
Files count | Total bundle size | % Changed
----------- | ----------------- | ---------
1 | 11.16 kB | 0%

<details>
<summary>View detailed bundle breakdown</summary>
<div>

**Added**
No assets were added

**Removed**
No assets were removed

**Bigger**
No assets were bigger

**Smaller**
No assets were smaller

**Unchanged**
Asset | File Size | % Changed
----- | --------- | ---------
index.js | 11.16 kB | 0%
</div>
</details>
</details>

### CODE REVIEW

File: .github/workflows/size-compare.yml [hunk 1]
```diff
@@ -47,75 +47,58 @@ jobs:
...
-                  );
-                }
-                core.info(`No successful ${label} build run yet — sleeping 15s.`);
-                await new Promise(r => setTimeout(r, SLEEP_MS));
-              }
-            }
-
+            const { resolveBuildRun } = await import(
+              `${process.env.GITHUB_WORKSPACE}/packages/ci-actions/src/build-runs.mjs`
+            );
```
[the remark this answers is not in the document]
Comment (tim-smart) [reply .github/workflows/size-compare.yml:65]: Addressed the valid revision-consistency issue in 6a86c0ceb by pinning checkout to `github.event.pull_request.base.sha`; regression coverage now verifies that trusted checkout occurs before the resolver import and never uses the PR head. I intentionally did not apply the proposed head checkout: this is a `pull_request_target` workflow with `pull-requests: write`, so importing fork-controlled code would create a pwn-request vulnerability. The workflow definition itself comes from the base revision, so this PR-added import is not executed before merge; once merged, the resolver exists at the trusted base SHA.

File: packages/ci-actions/src/build-runs.test.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import { readFile } from 'node:fs/promises';
+
+import { describe, expect, it, vi } from 'vitest';
+
+import { resolveBuildRun } from './build-runs.mjs';
```
[the remark this answers is not in the document]
Comment (tim-smart) [reply packages/ci-actions/src/build-runs.test.js]: Assessed against 229e2ba62 and kept the test as JavaScript. A mechanical `.ts` rename fails the package's strict typecheck with TS7016 because the directly executed `resolve.mjs` runtime module has no TypeScript declaration. Adding a hand-maintained declaration only to rename the test would duplicate the runtime contract without typechecking the resolver itself. The colocated JS test exercises the actual Node-loaded module, and all 19 tests plus the package typecheck pass.

File: packages/ci-actions/src/build-runs/size-compare-workflow.test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { readFile } from 'node:fs/promises';
+
+import { beforeAll, describe, expect, it } from 'vitest';
```
[the remark this answers is not in the document]
Comment (tim-smart) [reply packages/ci-actions/src/build-runs/size-compare-workflow.test.js]: Addressed in cbd8e9893: renamed the workflow-only regression test to `size-compare-workflow.test.ts` and typed `workflow` as `string`. Unlike `resolve.test.js`, this test does not import the untyped `.mjs` resolver, so strict package typechecking supports the conversion. The ci-actions tests and typecheck pass.

File: packages/ci-actions/src/build-runs/size-compare-workflow.test.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    const baseShaExpression = `\${{ github.event.pull_request.base.sha }}`;
+    const checkoutIndex = workflow.indexOf(`ref: ${baseShaExpression}`);
+    const resolverImportIndex = workflow.indexOf(
+      'packages/ci-actions/src/build-runs/resolve.mjs',
+    );
+
+    expect(checkoutIndex).toBeGreaterThan(-1);
+    expect(resolverImportIndex).toBeGreaterThan(checkoutIndex);
+    expect(workflow).toContain(`BASE_SHA: ${baseShaExpression}`);
+  });
```
[the remark this answers is not in the document]
Comment (tim-smart) [reply packages/ci-actions/src/build-runs/size-compare-workflow.test.js]: Addressed in cbd8e9893: the test now extracts the named `Checkout base revision`, `Set up environment`, and `Resolve build runs` step blocks. It verifies checkout-before-setup-before-resolver ordering and scopes the base-SHA ref, setup action, resolver import, and `BASE_SHA` assertions to their owning blocks. The ci-actions tests, typecheck, and repository lint pass.

**Review by MatissJanis — APPROVED** [review]

**Comment by tim-smart** [comment]
@coderabbitai resume

**Comment by tim-smart** [comment]
> Caution
> 
> Some comments are outside the diff and can’t be posted inline due to platform limitations.
> 
> ⚠️ Outside diff range comments (1)
> > .github/workflows/size-compare.yml (1)> `75-91`: _🩺 Stability & Availability_ | _🟠 Major_ | _⚡ Quick win_
> > > **Resolve the PR merge SHA, not the PR head SHA.**
> > > `build.yml` checks out the PR merge ref, and `resolveBuildRun` requires an exact `head_sha` match. Pass `github.event.pull_request.merge_commit_sha` instead of `github.event.pull_request.head.sha`; otherwise the lookup can time out. Add a test for this data flow.

GitHub Actions records pull_request runs under the PR head SHA even though the job checks out the merge ref, so switching to merge_commit_sha would match no run and cause the timeout described.
